Add a ComposeState for _wait_for_sync()

mattiaverga opened the pull request:
Fix #2550 

Signed-off-by: Mattia Verga <mattia.verga@tiscali.it>

mattiaverga committed:
Add a ComposeState for _wait_for_sync()

Signed-off-by: Mattia Verga <mattia.verga@tiscali.it>

bowlofeggs reviewed (changes requested):
Nice! This looks pretty good. The only thing we still need that I can think of is a database migration to extend the enum. There are a couple migrations that do something like this already that you can base your migration off of, for example:

https://github.com/fedora-infra/bodhi/blob/3.10.0/bodhi/server/migrations/versions/2c5d45f0c932_add_a_new_state_for_cleaning_composes.py

You can use alembic to start the template for you like this:

```
$ alembic revision --autogenerate -m "Add a syncing_repo ComposeState enum value."
```

mattiaverga commented:
I've added the `syncing_repo` value after `punging` instead at the end of the list to maintain the order specified in the docstring (where I added it in the right point of the sequence).
I'm totally new to alembic, but running the upgrade locally returned no errors, so I think it's ok...

mattiaverga commented:
jenkies test

bowlofeggs commented on bodhi/server/migrations/versions/68f9ccb1f388_add_a_syncing_repo_composestate_enum_.py:
```diff
@@ -0,0 +1,54 @@
+# Copyright (c) YEAR AUTHOR
```
Let's fill in the year and author:

```
# Copyright (c) 2018 Mattia Verga
```

bowlofeggs replied:
This is a minor enough of a change that I'll just force push your branch and fix it for you. Fix incoming!

bowlofeggs reviewed (changes requested):
Looks great - just need you to claim copyright on your migration and we should be good to merge.

bowlofeggs commented:
jenkies test

bowlofeggs commented:
jenkies test

bowlofeggs commented:
Mergify seems stuck, so I'm going to manually rebase and merge this one.
